src/sessions.ts

```diff
@@ -581,18 +611,34 @@ export class ClientSession
       bypassPinningCheck: true
     });
 
+    const timeoutMS =
+      typeof options?.timeoutMS === 'number'
+        ? options.timeoutMS
+        : this.timeoutContext?.csotEnabled()
+        ? this.timeoutContext.timeoutMS // refresh timeoutMS for abort operation
+        : typeof this.timeoutMS === 'number'
+        ? this.timeoutMS
+        : null;
+
+    const timeoutContext = TimeoutContext.create({ timeoutMS, ...this.clientOptions });
+
     try {
-      await executeOperation(this.client, operation);
+      await executeOperation(this.client, operation, timeoutContext);
       this.unpin();
       return;
     } catch (firstAbortError) {
       this.unpin();
 
+      if (options?.throwTimeout && MongoOperationTimeoutError.is(firstAbortError))
```

Comment:
According to this test: https://github.com/mongodb/node-mongodb-native/blob/main/test/spec/client-side-operations-timeout/sessions-inherit-timeoutMS.yml#L86-L125 

abortTransaction does not throw when called manually, but withTransaction should propagate a timeout error if one is encountered. I could be misunderstanding, but in order for that test to pass it would need to include either an expectError or an ignoreResultAndError flag.

Reply:
Is there a test that indicates that when an abortTransaction is inside of a convenient transaction it is supposed to throw?